content/blog/announcing-enum-support/index.md

```diff
@@ -0,0 +1,285 @@
+{{< /chooser >}}
+
+## Optimized for flexibility
+
+While some properties make sense as "strict" enums (i.e., the input value **must** be one of the enumerated values), there are times when it would be beneficial to have constants for commonly used values without restricting the input to **only** those values. We call this second category "relaxed" enums, and we model them a little differently in our schema.
+
+### "Strict" enums
+
+A property is a "strict" enum when the input value **must** be one of the enumerated values. In this case, the property type is specified as the enum type. Currently, only the Azure NextGen provider uses "strict" enums as the OpenAPI specification explicitly defines its properties as such.
+
+```json5
+{
+  "resources": {
+    "azure-nextgen:storage/latest:StorageAccount": {
+      "inputProperties": {
+        "accessTier": {
+          "$ref": "#/types/azure-nextgen:storage/latest:AccessTier"
+        },
+      },
+    },
+  },
+  "types": {
+    "azure-nextgen:storage/latest:AccessTier": {
+      "type": "string",
+      "enum": [
+        {
+          "value": "Hot"
+        },
+        {
+          "value": "Cool"
+        }
+      ],
+    },
+  },
+}
+```
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+**Note:** When using TypeScript, "strict" enums will accept both the constant (`AccessTier.Cool`) or the literal string (`"Cool"`). If there is an accidental spelling error in the literal, you are *immediately* alerted to the issue rather than having to wait until runtime.
+
+![ENUM_ERROR_TYPESCRIPT](ts-enum-spelling.png)
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+### "Relaxed" enums
+
+When a property is a "relaxed" enum, the property type is specified as the `Union` of the enum type and the underlying primitive type. This means that you have the convenience of using the enum constants (i.e., `s3.CannedAcl.Private`), but you may also pass in the raw string (i.e., `"private"`).
+
+In the AWS provider (and other Terraform-based providers), we have opted for **only** using "relaxed" enums. The reasoning here is twofold. Allowing the primitive type maintains backward compatibility and also allows users to use values that may not yet be represented in the Pulumi schema (e.g., a new Managed Policy ARN, EC2 Instance type, etc.).
+
+```json5
+{
+  "resources": {
+    "aws:s3/bucket:Bucket": {
+      "inputProperties": {
+        "acl": {
+          "oneOf": [
+            {
+              "type": "string"
+            },
+            {
+              "$ref": "#/types/aws:s3/CannedAcl:CannedAcl"
+            },
+          ]
+        },
+      },
+    },
+  },
+  "types": {
+    "aws:s3/CannedAcl:CannedAcl": {
+      "type": "string",
+      "enum": [
+        {
+          "name": "Private",
+          "value": "private"
+        },
+        ...
+      ],
+    },
+  },
+}
+```
+
+## Try them out!
+
+You can find enum types integrated into v3.19.0 of the AWS provider and v0.3.0 of the Azure NextGen provider.
```

Comment:
Perhaps cross link these (or another reference to the provider names) to the docs site?

```suggestion
You can find enum types integrated into `v3.19.0` of the AWS provider and `v0.3.0` of the Azure NextGen provider.
```